vault_test.go

```diff
@@ -0,0 +1,57 @@
+package main
+
+import (
+	"bytes"
+	"testing"
+)
+
+func TestGeneratePolicy(t *testing.T) {
+	w := new(bytes.Buffer)
+	info := &instanceInfo{
+		OrganizationGUID: "org-id",
+		SpaceGUID: "space-id",
+		ServiceInstanceGUID: "service-instance-id",
+		ApplicationGUID: "application-id",
+	}
+	if err := GeneratePolicy(w, info); err != nil {
+		t.Fatal(err)
+	}
+	result := w.String()
+	if result != expected {
+		t.Fatalf("received unexpected policy of %s", result)
+	}
+}
```

Comment:
Probably good to test the case with `ApplicationGUID == ""`.

Reply:
Yes, good point. Added!